apps/dynamo/lib/dynamo/vnode.ex

```diff
@@ -81,6 +105,35 @@ defmodule Vnode do
     end
   end
 
+  defp wait_read_response(key, value, index, context, state, current_read, num_read) do
+    if current_read < num_read do
+      receive do
+        {:ok, ^key, {other_value, other_context}, other_index} ->
+          cond do
+            VClock.compare_vclocks(context, other_context) == :after ->
+              Logger.info("vclock received. Result of comparison: After")
+              # Need to send a put request to the node behind
+              wait_read_response(key, value, index, context, state, current_read+1, num_read)
+            VClock.compare_vclocks(context, other_context) == :before ->
+              Logger.info("vclock received. Result of comparison: Before")
+              index_map = Map.get(state.data, key, %{})
+              index_map = Map.put(index_map, key, {other_value, other_context})
```

Comment:
Better to use [pipeline operator](https://elixir-lang.org/getting-started/enumerables-and-streams.html#eager-vs-lazy)
> index_map = 
state.data
|> Map.get(index, %{})
|> Map.put(key, {other_value, other_context})

Reply:
ok